### DISCUSSION TITLE
type: 'prompt' snap_dialog for password masking

### DISCUSSION THREAD

**Discussion by talken-devteam** [MetaMask/snaps#2491] [Ideas]
Hi I'm developing snap-simple-keyring wallet.
OnKeyringRequestHandler,
    const password = await snap.request({
      method: 'snap_dialog',
      params: {
        type: 'prompt',
        content: panel([
          heading('mpc password?'),
          text('Please input mpc password'),
        ]),
        placeholder: 'password...',
      },
    });
![image](https://github.com/MetaMask/snaps/assets/71870213/22cf49fd-5ba4-4e98-a07b-de38e37a20cd)
Is there a way to mask the input like inputType:"password"

**Comment by Montoya** [comment]
Hi, we are working on a customizable dialog that will provide more control over forms like these. You will be able to use JSX UI components with props, like `<Input type="password">`. We will have more information about this in a couple weeks.

**Comment by Montoya** [comment]
Hi @talken-devteam , please try the following: 

```
return snap.request({
  method: "snap_dialog",
  params: {
    // For this example we're using `DialogType.Confirmation`, but you can
    // also use the literal string value "confirmation" here.
    type: DialogType.Confirmation,
    content: panel([
      heading("Confirmation Dialog"),
      input({ name: "foo", inputType: "password" }),
    ]),
  },
});
```

**Reply by talken-devteam** [reply]
Hi thanks for your response sir. I tested like
```
    const password1 = await snap.request({
      method: 'snap_dialog',
      params: {
        type: 'prompt',
        content: panel([
          heading('mpc password1?'),
          text('mpc password1'),
        ]),
      },
    });
    console.log("password1::",password1)
    const password2 = await snap.request({
      method: "snap_dialog",
      params: {
        // For this example we're using `DialogType.Confirmation`, but you can
        // also use the literal string value "confirmation" here.
        type: DialogType.Confirmation,
        content: panel([
          heading("Confirmation Dialog"),
          input({ name: "foo", inputType: "password" }),
        ]),
      },
    });
    console.log("password2::",password2)
```
I can get password1 then I touch the input in Confirmation panel, get crash
Uncaught (in promise) Error: Cannot call write after a stream was destroyed.
I also tried in type: 'prompt' with input({ name: "foo", inputType: "password" }) then same error.

In @metamask\snaps-sdk\dist\types\types\methods\dialog.d.ts
/**
 * The result returned by the `snap_dialog` method.
 *
 * - If the dialog is an `alert`, the result is `null`.
 * - If the dialog is a `confirmation`, the result is a boolean indicating
 * whether the user confirmed the dialog.
 * - If the dialog is a `prompt`, the result is the value entered by
 * the user.
 */
confirmation returns boolean, prompt returns the value of text('') I think. I'd like to use prompt that returns input('') value like
```
    const password1 = await snap.request({
      method: 'snap_dialog',
      params: {
        type: 'prompt',
        content: panel([
          heading('mpc password1?'),
          //text('mpc password1'),// not this
          input({ name: "password1", inputType: "password" }),
        ]),
      },
    });
```
